app/controllers/room.controller.js

```diff
@@ -0,0 +1,44 @@
+import connect from '../../config/db';
+import Model from '../model/room';
+import User from '../model/user';
+
+export const create = (roomName, userName, socket) => {
+  connect.then(() => {
+    User.findOne({ name: userName })
+      .then((user) => {
+        Model.findOne({ roomName })
+          .then((rooms) => {
+            if (rooms) {
```

Comment:
I think it's bad code.
```
 if (rooms) {
              if (rooms.whiteList.includes(user._id)) {
                socket.join(roomName);
              } else {
                // TODO: implement functional
              }
            } else {
              const room = new Model({
                roomName,
                whiteList: user._id,
              });
              socket.join(roomName);
              return room.save();
            }
```
Move conditions into separate functions to make it clearer
```
rooms ? 
call function with (
if (rooms.whiteList.includes(user._id)) {
                socket.join(roomName);
              } else {
                // TODO: implement functional
              }
) 
: callElseFunction (  
 const room = new Model({
                roomName,
                whiteList: user._id,
              });
              socket.join(roomName);
              return room.save();
)
 
```